lib/api/apiUtils/object/versioning.js

```diff
@@ -129,10 +166,14 @@ export function versioningPreprocessing(bucketName, bucketMD, objectKey,
             if (nullVersionId === undefined) {
```

Comment:
when suspended, we always overwrite the master version by using the `''` option in the metadata put call. Normally this takes care of deleting/overwriting the previous null version metadata, as long as the null version is the master version.

However, if there was a previous null version in the version history and it is not the master version, we have to manually delete the null version metadata. Here we check if there is no null version before we do the manual delete.